diff -Nurp inotify-tools-2.2.orig/src/inotifywait.c inotify-tools-2.2/src/inotifywait.c
--- inotify-tools-2.2.orig/src/inotifywait.c	2006-07-03 06:49:34.000000000 +0200
+++ inotify-tools-2.2/src/inotifywait.c	2006-07-06 13:56:32.000000000 +0200
@@ -1,3 +1,4 @@
+#define _GNU_SOURCE
 #include "../config.h"
 
 #include <unistd.h>
@@ -17,7 +18,6 @@
 #include "inotify-syscalls.h"
 #include "inotifyutils.h"
 
-#define _GNU_SOURCE
 #include <getopt.h>
 
 extern char *optarg;
diff -Nurp inotify-tools-2.2.orig/src/inotifywatch.c inotify-tools-2.2/src/inotifywatch.c
--- inotify-tools-2.2.orig/src/inotifywatch.c	2006-07-03 06:49:34.000000000 +0200
+++ inotify-tools-2.2/src/inotifywatch.c	2006-07-06 13:57:40.000000000 +0200
@@ -112,7 +112,6 @@ int main(int argc, char ** argv)
 	if ( !events )
 		events = IN_ALL_EVENTS;
 
-	char * path;
 	unsigned int num_watches = 0;
 	unsigned int this_num_watches;
 	fprintf( stderr, "Establishing watches...\n" );
